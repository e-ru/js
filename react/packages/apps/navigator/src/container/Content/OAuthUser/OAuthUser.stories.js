import React from "react";
import { storiesOf } from "@storybook/react";

import { OAuthUserComponent } from "./OAuthUser";

storiesOf("OAuthUser", module)
  .add("with user", () => (
    <OAuthUserComponent
      oAuthUsers={[
        {
          id: 0,
          username: "test_user_1",
          email: "t@b.d",
          enabled: true,
          accountLocked: false,
          accountExpired: false,
        },
        {
          id: 1,
          username: "test_user_1",
          email: "q@b.d",
          enabled: true,
          accountLocked: false,
          accountExpired: false,
        },
      ]}
      match={{ params: { id: 0 } }}
      getUsersHanlder={() => console.log("getUsersHanlder clicked")}
      updateUserHandler={() => console.log("updateUserHandler clicked")}
    />
  ))
  .add("with no user", () => (
    <OAuthUserComponent
      oAuthUsers={[]}
      match={{ params: { id: 0 } }}
      getUsersHanlder={() => console.log("getUsersHanlder clicked")}
      updateUserHandler={() => console.log("updateUserHandler clicked")}
    />
  ));
