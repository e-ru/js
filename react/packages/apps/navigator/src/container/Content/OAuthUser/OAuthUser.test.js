import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { OAuthUserComponent } from "./OAuthUser";

configure({ adapter: new Adapter() });

function setup() {
  const props = {
    oAuthUsers: [],
    match: {
      params: { id: 0 },
    },
    getUsersHanlder: jest.fn(),
    updateUserHandler: jest.fn(),
  };

  const wrapper = shallow(<OAuthUserComponent {...props} />);

  return {
    wrapper,
  };
}

describe("<OAuthUserComponent />", () => {
  test("renders", () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
