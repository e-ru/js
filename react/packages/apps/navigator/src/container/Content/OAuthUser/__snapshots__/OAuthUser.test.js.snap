// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OAuthUserComponent /> renders 1`] = `ShallowWrapper {}`;
